massa-execution-worker/src/execution.rs

```diff
@@ -446,7 +445,11 @@ impl ExecutionState {
     ///
     /// # Returns
     /// An `ExecutionOutput` structure summarizing the output of the executed slot
-    pub fn execute_slot(&self, slot: Slot, opt_block: Option<(BlockId, Block)>) -> ExecutionOutput {
+    pub fn execute_slot(
+        &self,
+        slot: Slot,
+        opt_block: Option<(BlockId, &Block)>,
+    ) -> ExecutionOutput {
         // get optional block ID and creator address
         let (opt_block_id, opt_block_creator_addr) = opt_block
             .as_ref()
```

Comment:
With `execute_slot` now taking `Option<(BlockId, &Block)>` as argument, this `as_ref()` becomes unnecessary since `opt_block` won't get moved when retrieving the public key.

Reply:
Done.